### DISCUSSION TITLE
Spanish language by default

### DISCUSSION THREAD

**Discussion by SandraEliz** [open-sdg/open-sdg#1872] [Q&A] [answered]
Hello.
 
Please can you advise me on how to change the language to Spanish by default, I have not been able to make that change, it always marks errors.
 
Thank you

**Comment by brockfanning** [comment, marked as answer]
@SandraEliz I believe the challenge may be that when changing the languages, the automated tests will fail unless several files are changed at the same time. So, it is not possible to accomplish this by changing one file at a time in Github.com.

For example, if your first change is to modify the [languages site config setting](https://github.com/open-sdg/open-sdg-site-starter/blob/develop/_data/site_config.yml#L140-L142) then the "html-proofer" tests will be fail because the language-picker links will point to non-existent pages. But if you edit one of the pages first then automated tests will probably fail for different reasons.

The trick here is to edit all of the relevant files at the same time - or rather, in the same branch. There are two ways to accomplish this:

1. In Github.com
2. Locally using Git

There is also a third approach where you disable automated tests (temporarily).

I'll describe each below:

## 1. In Github.com

With this approach, you do the following:

1. Change any file, creating a pull-request as normal
2. Let the automated test fail, as expected
4. At the top of your pull-request look for "...wants to merge 1 commit into develop from..." which is followed by your branch name (eg, "brockfanning-patch-1".
5. Click on your branch name. This takes you to the root of the repository, but you will be "in" your branch.
6. Make sure that your are "in" your branch. You can confirm this by looking at the dropdown under "Code" in Github -- you should see your branch selected.
7. Now, navigate to a second file and edit it in the usual way. However when you are doing the "Commit changes" part, make sure you have selected "Commit directly to the `brockfanning-patch-1` branch" (or whatever yours is called)
8. Your automated tests will likely fail again, but you can keep repeating steps 5 and 6 above until finally you have edited all of the necessary files, and the automated tests finally pass.

## 2. Locally using Git

With this approach, you would need to have Git installed on your local computer, and you would need to be comfortable using Git. If you meet these requirements, then here are the basic steps:

1. Pull the latest "develop" code
2. Create a new branch for your changes
3. Change all the necessary files and commit your changes
4. Push your branch to Github and create a pull-request
5. Merge the pull-request as normal once all automated tests pass

## 3. Disabling automated tests

There is also another approach if there are challenges with the above. That is to actually disable automated tests (temporarily). To do this, you can try the following:

1. Go to "Settings" in Github
2. Select "Branches"
3. Press "Edit" next to the "develop" branch protection rule
4. Uncheck the box for "Require status checks to pass before merging"
5. Press "Save changes"

After this, you will be able to merge your pull-requests even when the tests fail.

NOTE: If you choose this approach, it is important to enable the automated tests again when you are finished:

1. Go to "Settings" in Github
2. Select "Branches"
3. Press "Edit" next to the "develop" branch protection rule
4. Check the box for "Require status checks to pass before merging"
5. Press "Save changes"

## Steps involved in changing the default language of the site starter

Also, here are all of the steps that are needed to change the site starter to default to Spanish, using Github.com. (As mentioned above, these need to be done in a single branch in order to avoid failing automated tests.)

1. Change the order of the [languages](https://github.com/open-sdg/open-sdg-site-starter/blob/develop/_data/site_config.yml#L140-L142) setting to:

        languages:
          - es
          - en
2. Change the `language` of the [test post](https://github.com/open-sdg/open-sdg-site-starter/blob/develop/_posts/2020-01-10-test-post-1.md) to "es":

        language: es
3. Change the `language` of the [other test post in the subfolder](https://github.com/open-sdg/open-sdg-site-starter/blob/develop/_posts/es/2020-01-10-test-post-1.md) to "en" and also add "/en" to the `permalink`:

        language: en
        permalink: /en/test-post-1/
4. Change the `language` of the [About page](https://github.com/open-sdg/open-sdg-site-starter/blob/develop/_pages/about.md) to "es":

        language: es
5. Change the `language` of the [other About page in the subfolder](https://github.com/open-sdg/open-sdg-site-starter/blob/develop/_pages/es/about.md) to "en", and also add "/en" to the `permalink`:

        language: en
        permalink: /en/about/
6. Repeat steps 4 and 5 for these other pages in the `_pages` folder and subfolder:
    * categories.md
    * contact-us.md
    * cookies.md
    * faq.md
    * news.md

7. At this point, the automated tests should be passing. But there are a couple more suggested steps.
8. For each of the files in the "es" subfolders that you edited above, you should also change the subfolder to "en". To do this, you edit each file and then change the filename. If you type BACKSPACE to delete the filename and then type BACKSPACE again, you can edit the folder from "es" to "en". You can then type "/" re-enter the filename as before (eg, "about.md"). Here is what that looks like:

   ![github-com-edit-file-folder](https://user-images.githubusercontent.com/1319083/201341778-eb7e6234-dacf-4630-a417-02e47acfe816.gif)

9. Perform Spanish translations of the `title` and content of each of the pages edited above, by editing the files as needed. You may first want to change the English content (the files that are now in the "en" subfolders) according to your preferences.

**Comment by brockfanning** [comment]
EDIT: I moved this guide up to the accepted answer for completeness.

Also, here are all of the steps that are needed to change the site starter to default to Spanish, using Github.com. (As mentioned above, these need to be done in a single branch in order to avoid failing automated tests.)

1. Change the order of the [languages](https://github.com/open-sdg/open-sdg-site-starter/blob/develop/_data/site_config.yml#L140-L142) setting to:

        languages:
          - es
          - en
2. Change the `language` of the [test post](https://github.com/open-sdg/open-sdg-site-starter/blob/develop/_posts/2020-01-10-test-post-1.md) to "es":

        language: es
3. Change the `language` of the [other test post in the subfolder](https://github.com/open-sdg/open-sdg-site-starter/blob/develop/_posts/es/2020-01-10-test-post-1.md) to "en" and also add "/en" to the `permalink`:

        language: en
        permalink: /en/test-post-1/
4. Change the `language` of the [About page](https://github.com/open-sdg/open-sdg-site-starter/blob/develop/_pages/about.md) to "es":

        language: es
5. Change the `language` of the [other About page in the subfolder](https://github.com/open-sdg/open-sdg-site-starter/blob/develop/_pages/es/about.md) to "en", and also add "/en" to the `permalink`:

        language: en
        permalink: /en/about/
6. Repeat steps 4 and 5 for these other pages in the `_pages` folder and subfolder:
    * categories.md
    * contact-us.md
    * cookies.md
    * faq.md
    * news.md

7. At this point, the automated tests should be passing. But there are a couple more suggested steps.
8. For each of the files in the "es" subfolders that you edited above, you should also change the subfolder to "en". To do this, you edit each file and then change the filename. If you type BACKSPACE to delete the filename and then type BACKSPACE again, you can edit the folder from "es" to "en". You can then type "/" re-enter the filename as before (eg, "about.md"). Here is what that looks like:

   ![github-com-edit-file-folder](https://user-images.githubusercontent.com/1319083/201341778-eb7e6234-dacf-4630-a417-02e47acfe816.gif)

9. Perform Spanish translations of the `title` and content of each of the pages edited above, by editing the files as needed. You may first want to change the English content (the files that are now in the "en" subfolders) according to your preferences.

**Comment by SandraEliz** [comment]
Hello.

I appreciate the guide regarding this issue, I was able to change it.

**Comment by SandraMarq** [comment]
Hello

When I change the default language to Spanish, I change the decimal point to a comma.

How can I make the change back to point?

Thank you

**Reply by sarah-beasley** [reply]
I've just seen in your site repository that you have added the line:
decimal_separator: '.' 
Did that not work?

**Reply by SandraEliz** [reply]
Hello

I put the decimal_separator: '.' in the _config.yml file, also in the site_config.yml file and it doesn't work, the comma keeps coming out.

**Reply by brockfanning** [reply]
Open SDG uses the javascript function [toLocaleString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString) to display numbers, and that will automatically change the decimal separator, regardless of the `decimal_separator` site config option. I recognize that the wording of our documentation on [decimal_separator](https://open-sdg.readthedocs.io/en/latest/configuration/#decimal_separator) makes it sound like decimal_separator will override the browser's choice. But that's not the case.

In theory we might be able to change our code such as decimal_separator -- if it is set -- overrides the toLocaleString function. However then it would not be different per language.

@SandraEliz It sounds like in your case the better language code would be "es-MX" - but then that would not work with our sdg-translations repository. We'll need to think on this issue and will get back to you. Thank you for bringing this to our attention!

**Reply by SandraMarq** [reply]
I am looking forward to your response

Thank you very much

**Reply by brockfanning** [reply]
I started a separate discussion #1887 to work on a solution to this.
